src/code/z_actor.c

```diff
@@ -457,439 +1231,3485 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+                                              actor->world.pos.y + (actor->shape.yOffset * actor->scale.y),
+                                              actor->world.pos.z, &actor->shape.rot);
+    }
+
+    Matrix_Scale(actor->scale.x, actor->scale.y, actor->scale.z, MTXMODE_APPLY);
+    Actor_SetObjectDependency(globalCtx, actor);
+
+    gSPSegment(POLY_OPA_DISP++, 0x06, globalCtx->objectCtx.status[actor->objBankIndex].segment);
+    gSPSegment(POLY_XLU_DISP++, 0x06, globalCtx->objectCtx.status[actor->objBankIndex].segment);
+
+    if (actor->colorFilterTimer != 0) {
+        s32 temp_v0_2 = actor->colorFilterParams & 0xC000;
+        Color_RGBA8 actorDefaultHitColor = { 0, 0, 0, 255 };
+
+        if (temp_v0_2 == 0x8000) {
+            actorDefaultHitColor.r = actorDefaultHitColor.g = actorDefaultHitColor.b =
+                ((actor->colorFilterParams & 0x1F00) >> 5) | 7;
+        } else if (temp_v0_2 == 0x4000) {
+            actorDefaultHitColor.r = ((actor->colorFilterParams & 0x1F00) >> 5) | 7;
+        } else if (temp_v0_2 == 0xC000) {
+            actorDefaultHitColor.b = actorDefaultHitColor.g = actorDefaultHitColor.r = 0;
+        } else {
+            actorDefaultHitColor.b = ((actor->colorFilterParams & 0x1F00) >> 5) | 7;
+        }
+
+        if (actor->colorFilterParams & 0x2000) {
+            func_800AE778(globalCtx, &actorDefaultHitColor, actor->colorFilterTimer, actor->colorFilterParams & 0xFF);
+        } else {
+            func_800AE434(globalCtx, &actorDefaultHitColor, actor->colorFilterTimer, actor->colorFilterParams & 0xFF);
+        }
+    }
+
+    actor->draw(actor, globalCtx);
+
+    if (actor->colorFilterTimer != 0) {
+        if (actor->colorFilterParams & 0x2000) {
+            func_800AE8EC(globalCtx);
+        } else {
+            func_800AE5A0(globalCtx);
+        }
+    }
+
+    if (actor->shape.shadowDraw != NULL) {
+        actor->shape.shadowDraw(actor, light, globalCtx);
+    }
+    actor->isDrawn = true;
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_800B9D1C(Actor* actor) {
+    s32 sfxId = actor->sfxId;
+
+    if (sfxId != 0) {
+        if (actor->audioFlags & 2) {
+            Audio_PlaySfxGeneral(sfxId, &actor->projectedPos, 4, &D_801DB4B0, &D_801DB4B0, &D_801DB4B8);
+        } else if (actor->audioFlags & 4) {
+            play_sound(sfxId);
+        } else if (actor->audioFlags & 8) {
+            func_8019F128(sfxId);
+        } else if (actor->audioFlags & 0x10) {
+            func_801A0810(&D_801DB4A4, NA_SE_SY_TIMER - SFX_FLAG, (sfxId - 1));
+        } else if (actor->audioFlags & 1) {
+            func_8019F1C0(&actor->projectedPos, sfxId);
+        }
+    }
+
+    if (sfxId) {}
+
+    if (actor->audioFlags & 0x40) {
+        // 0x27 is NA_BGM_MUSIC_BOX_HOUSE
+        func_801A1FB4(3, &actor->projectedPos, 0x27, 1500.0f);
+    }
+
+    if (actor->audioFlags & 0x20) {
+        // 0x71 is NA_BGM_KAMARO_DANCE
+        func_801A1FB4(0, &actor->projectedPos, 0x71, 900.0f);
+    }
```

Comment:
These bgm's can be filled in now

Reply:
I was thinking about this. Could we make a `bgmconvert.py` tool like `sfxconvert.py`? iirc, that tool relies on functions having a `sfxId` parameter, we could start naming the bgm parameter of those functions `bgmId`. What do you think?
(I know it is out of the scope of this PR, but I think it should be discussed)

Reply:
I imagine something like that would be pretty simple to make, so yeah that should be fine!